### DISCUSSION TITLE
location on android

### DISCUSSION THREAD

**Discussion by jroe1943** [artyom-beilis/skyhopper#29] [General]
I want to use old, but good, android phones with no sim card, hence no internet connectivity.  Maps uses the gps but AstroHopper doesn't seem to be able to do so, evidently a permission issue.  Is there any way to fix this?  Alternatively, or in addition, is there any way to manually input my lat and long as I really don't move around that much for observing.

**Comment by jroe1943** [comment]
I think I found a work around.  If I start up AstroHopper when my phone is connected via my home wifi and then turn on airplane mode, the GPS connection continues and I can travel out to my observing site with the app working.  I usually set the screen timeout to 30 minutes but turn if off before the timeout and restart it.  Works great.

I still think a manual entry of the lat and long would be a nice addition.

**Comment by artyom-beilis** [comment]
Yes it is something in todo list.

What happens if you open any GPS application and make sure it gets your location - I think at this point the phone should be aware of you location and thus astrohopper.

Can you check it?

By it is something I indeed need to fix.

**Reply by jroe1943** [reply]
Maps picks up location right away, location settings show that Maps and two others have permissions but AstroHopper does not show up as one of them ( it's not really an app though - but maybe Chrome needs the permission?).

Thanks for your work on this.  These days I'm trying to get more people into astronomy by designing a 3D printed telescope using a cheap Chinese mirror that can be built for under $100.  AstroHopper makes it much easier to use.
![AstroHopper](https://github.com/user-attachments/assets/f849e45b-1bdc-43d3-9344-67e19851632d)

**Comment by artyom-beilis** [comment]
https://github.com/artyom-beilis/skyhopper/issues/21

**Comment by jroe1943** [comment]
Chrome does have permissions to use location.
